grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -1846,6 +1849,9 @@ class JsonController {
                             destination             : it.destination,
                             productCode             : it.productCode,
                             productName             : it.productName,
+                            productFamily           : it.productFamily ?: '',
+                            category                : it.category ?: '',
+                            productCatalogs         : it.productCatalogs ?: '',
```

Comment:
shouldn't catalogs be done using `.join` here?
```suggestion
                            productCatalogs         : it.productCatalogs?.join(", "),
```

Reply:
I will add this in `ForecastingService` as I did it with `productCatalogs`

Reply:
Oh, I misunderstood your comment. I thought you said about `category`. I think there is no need to add `.join` here, because it is already added in `ForecastingService`

Reply:
Nevermind, I thought you were doing some two reports and haven't realized that the `data` is what is returned from the `ForecastingService` where it is already done. I'm sorry for confusion.

Reply:
So if that's the case, I think a better option would be to do whole "logic" inside the service, so to move `?: ''` to the service and here have only 
```
productFamily        : it.productFamily,
category             : it.category,

Reply:
+ if you do join and the list is empty, the result you get is an empty string, not null, so there is not need to do `?: ''` for `productCatalogs`